op-program/client/l2/engineapi/l2_engine_api.go

```diff
@@ -385,6 +385,10 @@ func (ea *L2EngineAPI) NewPayloadV4(ctx context.Context, params *eth.ExecutionPa
 		return &eth.PayloadStatusV1{Status: eth.ExecutionInvalid}, engine.UnsupportedFork.With(errors.New("newPayloadV4 called pre-isthmus"))
 	}
 
+	if params.WithdrawalsRoot == nil {
```

Comment:
Wondering if `L2EngineAPI.NewPayloadV4` has unit tests where we can add this.

Reply:
Indeed. It'll be good to add them here - https://github.com/ethereum-optimism/optimism/blob/b255e41ee96a038841927b819700bb6b6000488f/op-program/client/l2/engineapi/l2_engine_api_test.go#L24.

Reply:
Added some coverage here [cf4814d](https://github.com/ethereum-optimism/optimism/pull/14836/commits/cf4814dae46d6b10d242a5fe5f19898c6eaf02b9)